### ISSUE TITLE
GStreamer causes app crash after upgrading Android SDK from 22 to 23

### ISSUE THREAD

**Issue by liutingdu** [evercam/evercam-play-android#112]
Got different errors for different version of GStreamers: 

With GStreamer 1.4.5:
```

0java.lang.UnsatisfiedLinkError: dlopen failed: /data/app/io.evercam.androidapp-2/lib/arm/libgstreamer_android.so: has text relocations
1	at java.lang.Runtime.loadLibrary(Runtime.java:372)
2	at java.lang.System.loadLibrary(System.java:1076)
3	at io.evercam.androidapp.video.VideoActivity.&lt;clinit&gt;(VideoActivity.java:198)
4	at io.evercam.androidapp.custom.CameraLayout$1.onClick(CameraLayout.java:108)
```

With GStreamer 1.6.0:
```
17:23:10.827 26067-26067/? E/AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: library "./obj/local/armeabi/libgstreamer_android.so" not found
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at java.lang.Runtime.loadLibrary(Runtime.java:372)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at java.lang.System.loadLibrary(System.java:1076)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at io.evercam.androidapp.video.VideoActivity.<clinit>(VideoActivity.java:199)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at io.evercam.androidapp.custom.CameraLayout$1.onClick(CameraLayout.java:108)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at android.view.View.performClick(View.java:5198)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at android.view.View$PerformClick.run(View.java:21147)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at android.os.Handler.handleCallback(Handler.java:739)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:95)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at android.os.Looper.loop(Looper.java:148)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:5417)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at java.lang.reflect.Method.invoke(Native Method)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
10-16 17:23:10.827 26067-26067/? E/AndroidRuntime:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by liutingdu** [comment]
There is a similar issue http://stackoverflow.com/questions/32346402/libavcodec-so-has-text-relocations and the solutions are either keep using Android 22 or upgrade the GStreamer library. GStreamer can't get upgraded because of the existing issue https://github.com/evercam/evercam-play-android/issues/108 anyway, I rolled it back to Android 22 for now and will try this again when #108 gets fixed.

**Comment by liutingdu** [comment]
This is still happening after upgrading to GStreamer 1.6.1 with Android SDK 23. 

It's a [bug with GStreamer 1.6.1](https://bugzilla.gnome.org/show_bug.cgi?id=757732). I've did the quick fix in https://github.com/evercam/evercam-play-android/commit/50da951d68eda99bd7eb13a0152b3b11ef071d5b. The proper fix is to upgrade to 1.6.2, which has not been released yet, I'll review this after their next release.
